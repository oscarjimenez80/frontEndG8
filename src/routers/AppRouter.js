import React from 'react'
import { Routes, Route, BrowserRouter } from "react-router-dom";
import LoginPage from '../components/login/LoginPage';
import RegistraUsuario from '../components/usuarios/registro/RegistraUsuario';
import ContentRoutes from './ContentRoutes';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
            <Route path="/usuarios/crear" element={<RegistraUsuario />} />
            <Route path="/login" element={<LoginPage />} />
                <Route path="/" element={<LoginPage />} />

                <Route path="/*" element={
                    <PrivateRoute>
                        <ContentRoutes />
                    </PrivateRoute>
                } />

                {/* <Route path="/*" element={<ContentRoutes />} /> */}

            </Routes>
        </BrowserRouter>
    )
}

export default AppRouter
