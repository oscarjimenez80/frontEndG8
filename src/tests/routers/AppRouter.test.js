import { ApolloProvider } from "@apollo/client";
import { mount } from "enzyme";
import { AuthContext } from "../../provider/AuthProvider";
import AppRouter from "../../routers/AppRouter";

import client from '../../Apollo/client';



describe('Pruebas de <AppRouter />', () => {
    

    test('debe mostrar el login si no esta autenticado', () => {
        const contextValue = {
            isLogged() {
                return false;
            }
        }

        const wrapper = mount(
            <ApolloProvider client={client}>
                <AuthContext.Provider value={contextValue}>
                    <AppRouter />
                </AuthContext.Provider>
            </ApolloProvider>
        );

        // console.log(wrapper.html());

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h3').text()).toBe('Ingreso');

    })

    test('debe mostrar la pantalla de usuarios si esta autenticado', () => {
        const contextValue = {
            isLogged() {
                return true;
            }
        }

        const wrapper = mount(
            <ApolloProvider client={client}>
                <AuthContext.Provider value={contextValue}>
                    <AppRouter />
                </AuthContext.Provider>
            </ApolloProvider>
        );

        // console.log(wrapper.html());

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.navbar').exists()).toBe(true);
    })


})