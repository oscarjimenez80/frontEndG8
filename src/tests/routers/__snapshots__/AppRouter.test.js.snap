// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <AppRouter /> debe mostrar el login si no esta autenticado 1`] = `
<div
  className="login"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1 login-container"
    >
      <h3>
        Ingreso
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Correo"
            type="text"
          />
        </div>
        <div
          className="form-group mt-2"
        >
          <input
            className="form-control"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="ContraseÃ±a"
            type="password"
          />
        </div>
        <div
          className="form-group mt-3"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Pruebas de <AppRouter /> debe mostrar la pantalla de usuarios si esta autenticado 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Gestor de Proyectos
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/usuarios"
          onClick={[Function]}
        >
          Usuarios
        </a>
        <a
          className="nav-item nav-link "
          href="/proyectos"
          onClick={[Function]}
        >
          Proyectos
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Jhonny
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <section
    className="content mt-5"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <div
            className="card card-primary card-outline"
          >
            <div
              className="card-header flex"
            >
              <h5
                className="m-0"
              >
                Usuarios
              </h5>
              <a
                className="btn btn-primary mr-3"
                href="/usuarios/crear"
                onClick={[Function]}
              >
                Crear Usuario
              </a>
            </div>
            <div
              className="card-body"
            >
              <p>
                Cargando ...
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
